modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -1374,6 +1375,76 @@ func (suite *KeeperTestSuite) TestChanUpgradeCancel() {
 	}
 }
 
+// TestChanUpgrade_UpgradeSucceeds_AfterCancel verifies that if upgrade sequences
+// become out of sync, the upgrade can still be performed successfully after the upgrade is cancelled.
+func (suite *KeeperTestSuite) TestChanUpgrade_UpgradeSucceeds_AfterCancel() {
+	path := ibctesting.NewPath(suite.chainA, suite.chainB)
+	suite.coordinator.Setup(path)
+
+	path.EndpointA.ChannelConfig.ProposedUpgrade.Fields.Version = mock.UpgradeVersion
+	path.EndpointB.ChannelConfig.ProposedUpgrade.Fields.Version = mock.UpgradeVersion
+
+	suite.Require().NoError(path.EndpointA.ChanUpgradeInit())
+
+	// cause the upgrade to fail on chain b so an error receipt is written.
+	// if the counterparty (chain A) upgrade sequence is less than the current sequence, (chain B)
+	// an upgrade error will be returned by chain B during ChanUpgradeTry.
+	channel := path.EndpointA.GetChannel()
+	channel.UpgradeSequence = 1
+	path.EndpointA.SetChannel(channel)
+
+	channel = path.EndpointB.GetChannel()
+	channel.UpgradeSequence = 2
+	path.EndpointB.SetChannel(channel)
+
+	suite.Require().NoError(path.EndpointA.UpdateClient())
+	suite.Require().NoError(path.EndpointB.UpdateClient())
+
+	// error receipt is written to chain B here.
+	suite.Require().NoError(path.EndpointB.ChanUpgradeTry())
+
+	suite.Require().NoError(path.EndpointA.UpdateClient())
+
+	suite.T().Run("error receipt written", func(t *testing.T) {
+		_, ok := suite.chainB.GetSimApp().IBCKeeper.ChannelKeeper.GetUpgradeErrorReceipt(suite.chainB.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
+		suite.Require().True(ok)
+	})
+
+	suite.T().Run("upgrade cancelled successfully", func(t *testing.T) {
+		upgradeErrorReceiptKey := host.ChannelUpgradeErrorKey(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
+		errorReceiptProof, proofHeight := suite.chainB.QueryProof(upgradeErrorReceiptKey)
+
+		errorReceipt, ok := suite.chainB.GetSimApp().IBCKeeper.ChannelKeeper.GetUpgradeErrorReceipt(suite.chainB.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
```

Comment:
maybe you could just use the error receipt from earlier also, since it shouldn't have changed?